app/controllers/api/v1x0/mixins/rbac_mixin.rb

```diff
@@ -3,9 +3,11 @@ module V1x0
     module Mixins
       module RBACMixin
         def write_access_check
-        return unless RBAC::Access.enabled?
-        access_obj = RBAC::Access.new(controller_name.classify.constantize.table_name, 'write').process
-        raise Catalog::NotAuthorized, "Write access not authorized for #{controller_name.classify.constantize}" unless access_obj.accessible?
+          return unless RBAC::Access.enabled?
+          access_obj = RBAC::Access.new(controller_name.classify.constantize.table_name, 'write').process
+          raise Catalog::NotAuthorized, "Write access not authorized for #{controller_name.classify.constantize}" unless access_obj.accessible?
+          ids = access_obj.id_list
+          raise Catalog::NotAuthorized, "Write access not authorized for #{controller_name.classify.constantize}" if ids.any? && ids.exclude?(params[:id])
```

Comment:
The tests seem to convey that having a list of `ids` is what we want. The above code says to `raise` an error if we have ids and those ids don't include `params[:id]`. It seems the test cases are saying the opposite. Do you mean to have `unless` instead or are the tests incorrect?

Reply:
@syncrou Presence of ids makes it restrictive, so for Administrators you will get back an empty list which indicates no filtering needs to happen. But for users you might say that they have restricted access and are only allowed to look at specific ids.

Reply:
Never mind the above comment. I see you only raise if both parts are true `&&`.

Reply:
So, in the case of `ids = []` that is equal to '*' in an ACL?

Reply:
Yes